### ISSUE TITLE
Question cuid include `-` character?

### ISSUE THREAD

**Issue by odk211** [ericelliott/cuid#127]
It seems `-` character is included in cuid.(v 2.1.5)

```
> cuid = require('cuid')
> cuid()
'cjrvpx18e001fyax6dw11-pgw'
```

I check the [README why-are-there-no-dashes](https://github.com/ericelliott/cuid#why-are-there-no-dashes), so I wonder is it the expected behavior of cuid?

my nodejs version is v10.15.1

**Comment by didac-wh** [comment]
Not 100% sure of my words, but at least 1 package that uses cuid as dependency has recently (since today) started failing.
https://www.npmjs.com/package/yamlinc

Some includes are not working (and I saw that all the not working includes contain the '-' character).

I don't know if more packages may be affected by this, but I'd say this change would apply to a major release.

I opened an issue in yamlinc's github:
https://github.com/javanile/yamlinc/issues/16

Hope it helps, thank you.

**Comment by ericelliott** [comment]
I don't know that we have explicitly disallowed dashes, but I don't remember seeing them before in cuids. I would argue that dashes probably should NOT cause problems for projects depending on ID strings, but since they are, I'm open to a PR. Thank you for bringing this to my attention.

**Comment by ericelliott** [comment]
AFAIK, nothing in cuid changed that would suddenly allow dashes, and I never tried to disallow them in the first place. In other words, I don't think the recent updates changed cuid's behavior in that regard. I do see that it does generate an id with a dash about 5-6 percent of the time.

**Comment by ericelliott** [comment]
Looks like it did not generate dashes in v2.1.4. I stand corrected.

**Comment by ericelliott** [comment]
It looks like it may have been introduced by the change to use the crypto API instead of `Math.random()` (this is the only significant change I see to how IDs are generated). I'm open to a fix that replaces `-` characters with something that was already in the character set.

**Comment by ericelliott** [comment]
In the meantime, if this is a breaking change for you, you can fix it by pinning your cuid require to `cuid@2.1.4`.

**Comment by francescobianco** [comment]
Hi @ericelliott 

my understanding is this: 
using big number complementation can switch number sign from + to - (minus)

https://github.com/ericelliott/cuid/blob/c5e454fb1e7929ada4d40248bb36ca6015c8f293/lib/getRandomValue.js#L4

the before output can be

cuid() = cjrvpx18e001fyax6      dw11     -pgw
                                                 ^            ^
----------------------------------------/rand-----/rand (with minus)

**Comment by ericelliott** [comment]
Yep. That looks like the culprit. If we wrap `Math.abs` around the returned value, that should fix it.

```js
return Math.abs(crypto.randomBytes(4)
        .readInt32BE() / lim);
```

Want to make a PR?

**Comment by ericelliott** [comment]
PR open. Feel free to review and comment.

**Comment by ericelliott** [comment]
Published `cuid@2.1.6`
